### PR TITLE
SPUI-1005 > Tap states for some elements

### PR DESCRIPTION
**Pull request by Anton-Tesenchuk**

### CODE REVIEW

File: plugins/simple_kit/lib/example/buttons/examples/simple_primary_button_example.dart [hunk 1]
```diff
@@ -31,6 +31,12 @@ class SimplePrimaryButtonExample extends StatelessWidget {
                 onTap: () {},
               ),
               const SpaceH20(),
+              SPrimaryButton4(
```
Comment (eli1stark) [inline plugins/simple_kit/lib/example/buttons/examples/simple_primary_button_example.dart:34]: delete

File: lib/auth/screens/onboarding/onboarding_screen.dart [hunk 2]
```diff
@@ -176,7 +176,7 @@ class _OnBoardingScreenState extends State<OnboardingScreen>
                     onTap: () => Register.push(context),
                   ),
                   const SpaceH10(),
-                  STextButton1(
+                  SPrimaryButton4(
```
Comment (eli1stark) [inline lib/auth/screens/onboarding/onboarding_screen.dart:179]: revert

File: plugins/simple_kit/lib/src/buttons/basic_buttons/primary_button/public/simple_primary_button_4.dart [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+import '../../../../current_theme_stpod.dart';
+import '../dark/simple_dark_primary_button_3.dart';
+import '../light/simple_light_primary_button_4.dart';
+
+class SPrimaryButton4 extends ConsumerWidget {
```
Comment (eli1stark) [inline plugins/simple_kit/lib/src/buttons/basic_buttons/primary_button/public/simple_primary_button_4.dart:8]: delete this file

File: plugins/simple_kit/lib/src/shared/components/simple_show_alert_popup.dart [hunk 4]
```diff
@@ -87,11 +87,17 @@ void sShowAlertPopup(
...
                       )
-                    else
+                    else if (primaryButtonType == SButtonType.primary3)
                       SPrimaryButton3(
                         name: primaryButtonName,
                         active: activePrimaryButton,
                         onTap: () => onPrimaryButtonTap(),
+                      )
+                    else
+                      SPrimaryButton4(
```
Comment (eli1stark) [inline plugins/simple_kit/lib/src/shared/components/simple_show_alert_popup.dart:97]: revert

File: plugins/simple_kit/lib/src/buttons/resend_button/simple_resend_button.dart [hunk 5]
```diff
@@ -35,7 +35,7 @@ class SResendButton extends StatelessWidget {
           maintainSize: true,
           maintainAnimation: true,
           maintainState: true,
-          child: STextButton1(
+          child: SPrimaryButton4(
```
Comment (eli1stark) [inline plugins/simple_kit/lib/src/buttons/resend_button/simple_resend_button.dart:38]: revert

File: plugins/simple_kit/lib/src/shared/models/simple_button_type.dart [hunk 6]
```diff
@@ -2,4 +2,5 @@ enum SButtonType {
   primary1,
   primary2,
   primary3,
+  primary4,
```
Comment (eli1stark) [inline plugins/simple_kit/lib/src/shared/models/simple_button_type.dart:5]: delete

**Review by eli1stark — CHANGES_REQUESTED** [review]

**Comment by eli1stark** [comment]
The main task was to update SBaseTextButton and STextButton1 and STextButton2. But you create new primary component

File: plugins/simple_kit/lib/src/buttons/basic_buttons/primary_button/light/simple_light_primary_button_4.dart [hunk 7]
```diff
@@ -0,0 +1,34 @@
+import 'package:flutter/material.dart';
+
+import '../../../../colors/view/simple_colors_light.dart';
+import '../base/simple_base_primary_button.dart';
+
+class SimpleLightPrimaryButton4 extends StatelessWidget {
```
[the remark this answers is not in the document]
Comment (Anton-Tesenchuk) [reply plugins/simple_kit/lib/src/buttons/basic_buttons/primary_button/light/simple_light_primary_button_4.dart]: > delete this file

done

File: plugins/simple_kit/lib/src/buttons/basic_buttons/primary_button/dark/simple_dark_primary_button_3.dart [hunk 8]
```diff
@@ -0,0 +1,34 @@
+import 'package:flutter/material.dart';
+
+import '../../../../colors/view/simple_colors_dark.dart';
+import '../base/simple_base_primary_button.dart';
+
+class SimpleDarkPrimaryButton3 extends StatelessWidget {
```
[the remark this answers is not in the document]
Comment (Anton-Tesenchuk) [reply plugins/simple_kit/lib/src/buttons/basic_buttons/primary_button/dark/simple_dark_primary_button_3.dart]: > delete this file

done

File: plugins/simple_kit/lib/src/buttons/basic_buttons/primary_button/base/simple_base_primary_button.dart [hunk 9]
```diff
@@ -34,7 +37,9 @@ class SimpleBasePrimaryButton extends HookWidget {
 
     if (active) {
       if (highlighted.value) {
-        currentColor = activeColor.withOpacity(0.8);
+        currentColor = isTransparent != null
```
[the remark this answers is not in the document]
Comment (Anton-Tesenchuk) [reply plugins/simple_kit/lib/src/buttons/basic_buttons/primary_button/base/simple_base_primary_button.dart]: > revert changes

done

File: plugins/simple_kit/lib/src/buttons/basic_buttons/primary_button/base/simple_base_primary_button.dart [hunk 10]
```diff
@@ -17,6 +19,7 @@ class SimpleBasePrimaryButton extends HookWidget {
   }) : super(key: key);
 
   final Widget? icon;
+  final bool? isTransparent;
```
[the remark this answers is not in the document]
Comment (Anton-Tesenchuk) [reply plugins/simple_kit/lib/src/buttons/basic_buttons/primary_button/base/simple_base_primary_button.dart]: > delete

done

File: plugins/simple_kit/lib/src/buttons/basic_buttons/primary_button/base/simple_base_primary_button.dart [hunk 11]
```diff
@@ -1,12 +1,14 @@
...
 import 'package:flutter_hooks/flutter_hooks.dart';
 
+import '../../../../colors/view/simple_colors_dark.dart';
 import '../../base/simple_base_button.dart';
 
 class SimpleBasePrimaryButton extends HookWidget {
   const SimpleBasePrimaryButton({
     Key? key,
     this.icon,
+    this.isTransparent,
```
[the remark this answers is not in the document]
Comment (Anton-Tesenchuk) [reply plugins/simple_kit/lib/src/buttons/basic_buttons/primary_button/base/simple_base_primary_button.dart]: > delete

done

File: plugins/simple_kit/lib/example/buttons/examples/simple_primary_button_example.dart (hunk 1 above)
Comment (Anton-Tesenchuk) [reply plugins/simple_kit/lib/example/buttons/examples/simple_primary_button_example.dart]: > delete

done

File: lib/auth/screens/onboarding/onboarding_screen.dart (hunk 2 above)
Comment (Anton-Tesenchuk) [reply lib/auth/screens/onboarding/onboarding_screen.dart]: > revert

done

File: plugins/simple_kit/lib/example/buttons/examples/simple_primary_button_example.dart [hunk 12]
```diff
@@ -42,6 +48,12 @@ class SimplePrimaryButtonExample extends StatelessWidget {
                 name: 'Primary',
                 onTap: () {},
               ),
+              const SpaceH20(),
```
[the remark this answers is not in the document]
Comment (Anton-Tesenchuk) [reply plugins/simple_kit/lib/example/buttons/examples/simple_primary_button_example.dart]: > delete

done

File: plugins/simple_kit/lib/example/buttons/examples/simple_primary_button_example.dart [hunk 13]
```diff
@@ -31,6 +31,12 @@ class SimplePrimaryButtonExample extends StatelessWidget {
                 onTap: () {},
               ),
               const SpaceH20(),
+              SPrimaryButton4(
+                active: true,
+                name: 'Primary',
+                onTap: () {},
+              ),
+              const SpaceH20(),
```
[the remark this answers is not in the document]
Comment (Anton-Tesenchuk) [reply plugins/simple_kit/lib/example/buttons/examples/simple_primary_button_example.dart]: > delete

done

File: plugins/simple_kit/lib/src/buttons/basic_buttons/primary_button/public/simple_primary_button_4.dart (hunk 3 above)
Comment (Anton-Tesenchuk) [reply plugins/simple_kit/lib/src/buttons/basic_buttons/primary_button/public/simple_primary_button_4.dart]: > delete this file

done

File: plugins/simple_kit/lib/src/shared/components/simple_show_alert_popup.dart (hunk 4 above)
Comment (Anton-Tesenchuk) [reply plugins/simple_kit/lib/src/shared/components/simple_show_alert_popup.dart]: > revert

done

File: plugins/simple_kit/lib/src/buttons/resend_button/simple_resend_button.dart (hunk 5 above)
Comment (Anton-Tesenchuk) [reply plugins/simple_kit/lib/src/buttons/resend_button/simple_resend_button.dart]: > revert

done

File: plugins/simple_kit/lib/src/shared/components/simple_show_alert_popup.dart [hunk 14]
```diff
@@ -87,11 +87,17 @@ void sShowAlertPopup(
                         active: activePrimaryButton,
                         onTap: () => onPrimaryButtonTap(),
                       )
-                    else
+                    else if (primaryButtonType == SButtonType.primary3)
```
[the remark this answers is not in the document]
Comment (Anton-Tesenchuk) [reply plugins/simple_kit/lib/src/shared/components/simple_show_alert_popup.dart]: > revert

done

File: plugins/simple_kit/lib/src/shared/models/simple_button_type.dart (hunk 6 above)
Comment (Anton-Tesenchuk) [reply plugins/simple_kit/lib/src/shared/models/simple_button_type.dart]: > delete

done

**Comment by Anton-Tesenchuk** [comment]
> The main task was to update SBaseTextButton and STextButton1 and STextButton2. But you create new primary component

done

**Pull request merged**